src/main/java/backend/RepoID.java

```diff
@@ -0,0 +1,83 @@
+package backend;
+
+import org.eclipse.egit.github.core.RepositoryId;
+
+/**
+ * Represents the ID of a repository. To be used to ensure safe comparison
+ */
+public final class RepoID {
+
+    private final String repoOwner;
+    private final String repoName;
+
+    /**
+     * Constructor
+     * @param repoIDString repoOwner and repoName separated by slash
+     * @throws IllegalArgumentException If repoISString is invalid
+     */
+    public RepoID(String repoIDString) throws IllegalArgumentException {
+        boolean isValidRepoID = RepoID.isWellFormedRepoId(repoIDString);
+        if (isValidRepoID) {
+            String[] repoIDComponents = repoIDString.toLowerCase().split("/");
+            this.repoOwner = repoIDComponents[0];
+            this.repoName = repoIDComponents[1];
+        } else {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    /**
+     * Constructor
+     * @param repoOwner Owner of the repo
+     * @param repoName Name of the repo
+     * @throws IllegalArgumentException If repoOwner or repoName is invalid
+     */
+    public RepoID(String repoOwner, String repoName) throws IllegalArgumentException {
+        boolean isValidRepoID = RepoID.isWellFormedRepoId(repoOwner, repoName);
+        if (isValidRepoID) {
+            this.repoOwner = repoOwner.toLowerCase();
+            this.repoName = repoName.toLowerCase();
+        } else {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    public static boolean isWellFormedRepoId(String owner, String repo) {
```

Comment:
This method doesn't seem to be doing what it advertises. If it was, it shouldn't need the `owner` parameter.